### PR TITLE
Disable USM pooling tests while implementation is being changed.

### PR DESCRIPTION
**Pull request by rdeodhar**
This change removes tests previously added for USM pooling. The pooling implementation is being changed. Once that change is completed these tests will be adjusted and reenabled.

### CODE REVIEW

**Comment by rdeodhar** [comment]
The two unexpected passes are expected. A change planned in the intel_llvm repo will cause those tests to fail. That's why they are marked ax XFAIL now, in anticipation of the intel_llvm change.

**Review by sergey-semenov — DISMISSED** [review]

**Review by jbrodman — DISMISSED** [review]

**Review by vladimirlaz — APPROVED** [review]

File: SYCL/USM/usm_leak_check.cpp
Comment (vladimirlaz) [inline SYCL/USM/usm_leak_check.cpp]: Only SYCL :: USM/usm_pooling.cpp failed on windows (see testing in intel/llvm#4799). The current test should not be XFAILED there

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/USM/usm_leak_check.cpp
Comment (rdeodhar) [reply SYCL/USM/usm_leak_check.cpp]: Windows is currently **unsupported** by usm_leak_check.cpp because the ZE_DEBUG=4 feature that the test depends on does not work on Windows. See line 3 in the test. So the XFAIL takes effect on Linux only. Yet, it should do no harm on Windows.
Otherwise, please suggest a combination of UNSUPPORTED, XFAIL, REQUIRES that you think would be more appropriate.

Comment (vladimirlaz) [reply SYCL/USM/usm_leak_check.cpp]: in this case, it should be ok. UNSUPORTED disables the test and XFAIL should be ignored

**Review by vladimirlaz — DISMISSED** [review]

**Comment by rdeodhar** [comment]
Nothing has been changed, but need re-approval from code owners.

**Review by sergey-semenov — DISMISSED** [review]

**Review by jbrodman — DISMISSED** [review]

**Review by againull — APPROVED** [review]

**Review by jbrodman — APPROVED** [review]

**Comment by againull** [comment]
Unrelated ESIMD tests failed, merging.

**Pull request merged**
